engines/wintermute/ad/ad_actor_3dx.cpp

```diff
@@ -0,0 +1,2490 @@
+			PersistMgr->transferSint32(TMEMBER(NumItems));
+			for (uint32 i = 0; i < _transitionTimes.size(); i++) {
+				_transitionTimes[i]->Persist(PersistMgr);
+			}
+		} else {
+			int NumItems = _transitionTimes.size();
+			PersistMgr->transferSint32(TMEMBER(NumItems));
+			for (int i = 0; i < NumItems; i++) {
+				BaseAnimationTransitionTime *Trans = new BaseAnimationTransitionTime();
+				Trans->Persist(PersistMgr);
+				_transitionTimes.add(Trans);
+			}
+		}
+	}
+
+	if (PersistMgr->checkVersion(1, 7, 92)) {
+		PersistMgr->transferSint32(TMEMBER(_talkAnimChannel));
+	} else {
+		_talkAnimChannel = 0;
+	}
+
+	if (PersistMgr->checkVersion(1, 8, 4)) {
+		PersistMgr->transferString(TMEMBER(_partBone));
+		PersistMgr->transferVector3d(TMEMBER(_partOffset));
+	} else {
+		_partOffset = Math::Vector3d(0, 0, 0);
+	}
+
+	return true;
+}
+
+//////////////////////////////////////////////////////////////////////////
+bool AdActor3DX::invalidateDeviceObjects() {
+	if (_modelX)
+		_modelX->invalidateDeviceObjects();
+	if (_shadowModel)
+		_shadowModel->invalidateDeviceObjects();
+
+	for (uint32 i = 0; i < _attachments.size(); i++) {
+		_attachments[i]->invalidateDeviceObjects();
+	}
+
+	return true;
+}
+
+//////////////////////////////////////////////////////////////////////////
+bool AdActor3DX::restoreDeviceObjects() {
+	if (_modelX) {
+		_modelX->restoreDeviceObjects();
+	}
+
+	if (_shadowModel) {
+		_shadowModel->restoreDeviceObjects();
+	}
+
+	for (uint32 i = 0; i < _attachments.size(); i++) {
+		_attachments[i]->restoreDeviceObjects();
+	}
+
+	return true;
+}
+
+//////////////////////////////////////////////////////////////////////////
+bool AdActor3DX::mergeAnimations(const char *Filename) {
+	if (!_modelX) {
+		return false;
+	}
+
+	warning("AdActor3DX::mergeAnimations merging of .X is not implemented");
+
+	bool res = false;
+	if (!res) {
+		_gameRef->LOG(res, "Error: MergeAnims failed for file '%s'", Filename);
+		return res;
+	}
+
+	Common::String animExtFile = PathUtil::getFileNameWithoutExtension(Filename);
+	animExtFile += ".anim";
+
+	Common::SeekableReadStream *testFile = BaseFileManager::getEngineInstance()->openFile(animExtFile);
+
+	if (testFile) {
+		BaseFileManager::getEngineInstance()->closeFile(testFile);
+		return mergeAnimations2(animExtFile.c_str());
+	} else {
+		return true;
+	}
+}
+
+//////////////////////////////////////////////////////////////////////////
+bool AdActor3DX::mergeAnimations2(const char *Filename) {
+	TOKEN_TABLE_START(commands)
+		TOKEN_TABLE(ANIMATION)
+	TOKEN_TABLE_END
+
+	byte *buffer = BaseFileManager::getEngineInstance()->readWholeFile(Filename);
+	if (buffer == nullptr) {
+		return false;
+	}
+
+	byte *BufferOrig = buffer;
+
+	byte *params;
+	int cmd;
+	BaseParser parser;
+
+	while ((cmd = parser.getCommand((char **)&buffer, commands, (char **)&params)) > 0) {
+		switch (cmd) {
+		case TOKEN_ANIMATION: {
+			if (!_modelX->parseAnim(params)) {
+				cmd = PARSERR_GENERIC;
+			}
+		}
+			break;

```

Comment:
what is going on with this break? should be part of 'case'

Reply:
It is, but the content of the case statement is wrapped in braces. I suppose it could be put inside the braces, though

Reply:
maybe not put into switch at all. it doesn't have default, some compiler complains, or analysers

Reply:
Will do, but there are some more similar cases, where I put the break inside the braces to prevent it from looking too offplaced

Reply:
yes, put into break inside braces